src/main/java/seedu/bookmark/model/ModelManager.java

```diff
@@ -142,9 +173,62 @@ public void updateFilteredBookList(Predicate<Book> predicate) {
 
     @Override
     public void sortFilteredBookList(Comparator<Book> comparator) {
+        if (comparator instanceof BookNameComparator) {
+            setSortingPreference(PREFIX_NAME.toString());
+        } else if (comparator instanceof BookGenreComparator) {
+            setSortingPreference(PREFIX_GENRE.toString());
+        } else if (comparator instanceof BookPagesReadComparator) {
+            setSortingPreference(PREFIX_BOOKMARK.toString());
+        } else if (comparator instanceof BookReadingProgressComparator) {
+            setSortingPreference(PREFIX_READING_PROGRESS.toString());
+        }
+        this.comparator = comparator;
         this.library.sortBooks(comparator);
     }
 
+    /**
+     * Returns a comparator based on the input prefix
+     * @return Comparator based on input prefix
+     */
+    public Comparator<Book> comparatorGenerator(Prefix inputPrefix) {
```

Comment:
Noticed that this exact method is repeated inside `SortCommandParser` also. Can think about abstracting it out as a static utility method, maybe smth like `BookComparatorGenerator#comparatorGenerator(Prefix inputPrefix)`